src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2373,10 +2370,60 @@ private void UpperBoundTypeArgumentInference(NamedTypeSymbol source, NamedTypeSy
         //
         private bool Fix(int iParam, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            // UNDONE: This method makes a lot of garbage.
-
             Debug.Assert(IsUnfixed(iParam));
 
+            var exact = _exactBounds[iParam];
+            var lower = _lowerBounds[iParam];
+            var upper = _upperBounds[iParam];
+
+            // Fix considering nullability, if possible, otherwise fix ignoring nullability.
+            // PROTOTYPE(NullableReferenceTypes): Avoid calling Fix ignoring nullability if the nullability call succeeds.
+
+            HashSet<DiagnosticInfo> ignoredDiagnostics = null;
+            var withNullability = Fix(exact, lower, upper, ref ignoredDiagnostics, _conversions, includeNullability: true);
+            var withoutNullability = Fix(exact, lower, upper, ref useSiteDiagnostics, _conversions, includeNullability: false);
```

Comment:
What breaks if this second phase (without nullability) was removed?

Reply:
If we remove the `withoutNullability` pass, we'll have type inference failures in cases where type arguments are convertible when ignoring nullability. We should report a nullability mismatch warning in those cases rather a type inference error.
```
class C<T, U> { }
class C
{
    static void F<T>(T x, T y) { }
    static void G(C<string, object?> x, C<string?, object> y)
    {
        F(x, y);
    }
}
```
```
(7,14): warning CS8620: Nullability of reference types in argument of
    type 'C<string?, object>' doesn't match target type 'C<string, object?>'
    for parameter 'y' in
    'void C.F<C<string, object?>>(C<string, object?> x, C<string, object?> y)'.
```